src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/HelpersWindows.cs

```diff
@@ -327,6 +327,12 @@ public static AlgorithmIdentifier ToAlgorithmIdentifier(this CRYPT_ALGORITHM_IDE
             }
 
             AlgorithmIdentifier algorithmIdentifier = new AlgorithmIdentifier(Oid.FromOidValue(oidValue, OidGroup.All), keyLength);
+            switch (oidValue)
+            {
+                case Oids.RsaOaep:
```

Comment:
I was trying to decide if this should be opt-in or opt-out.  I guess opt-in makes enough sense, the recipient is going to need to have their own check for the algorithm ID, and as long as they add tests they'll realize that something's missing.

Reply:
My usual approach is "least impact". Since we were never populating parameters in the first place, and we only need them now for OAEP, my logic was to only populate the parameters for the necessary case.